llama.cpp

```diff
@@ -7783,6 +7783,73 @@ void llama_sample_typical(struct llama_context * ctx, llama_token_data_array * c
     }
 }
 
+void llama_sample_entropy(struct llama_context * ctx, llama_token_data_array * candidates_p, float min_temp, float max_temp, float exponent_val) {
+    const int64_t t_start_sample_us = ggml_time_us();
+
+    llama_sample_softmax(ctx, candidates_p);
```

Comment:
```suggestion
    llama_sample_softmax(nullptr, candidates_p);
```

Otherwise, this will count the `llama_sample_softmax` time 2 times